### PR TITLE
feat(flask): set response and request headers for auto-instrumented

### PR DESCRIPTION
**Pull request by majorgreys**
## Description

Set response and request headers for auto-instrumented Flask applications, as added for middleware in #1874.

Also make use of error status codes in #1860 and deprecate existing Flask-specific solution.

## Checklist
- [x] Entry added to release notes using [reno](https://docs.openstack.org/reno/latest/user/usage.html)
- [x] Tests provided; and/or
- [ ] Description of manual testing performed and explanation is included in the code and/or PR.
- [x] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: ddtrace/contrib/flask/patch.py [hunk 1]
```diff
@@ -307,11 +307,19 @@ def traced_start_response(status_code, headers):
...
-                s.set_tag(http.STATUS_CODE, code)
-                if 500 <= code < 600:
-                    s.error = 1
-                elif code in config.flask.get("extra_error_codes", set()):
-                    s.error = 1
+                trace_utils.set_http_meta(s, config.flask, status_code=code, response_headers=headers)
+
+                extra_error_codes = config.flask.get("extra_error_codes")
+                if extra_error_codes:
+                    log.warning(
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/flask/patch.py:314]: We should probably do a `debtcollector.deprecate()` here instead. We can probably set a `removal_version` to an upcoming version.

File: releasenotes/notes/flask-http-helpers-0167071f46a36e00.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+deprecations:
+  - |
+    flask: Use :ref:`HTTP Custom Error Codes <http-custom-error>` instead of ``ddtrace.config.flask['extra_error_codes']``.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/flask-http-helpers-0167071f46a36e00.yaml:4]: Does this link actually work when we build the docs? (Not sure if the RST gets rendered or not)

**Review by Kyle-Verhoog — COMMENTED** [review]
lgtm! Just a question and a suggestion

File: releasenotes/notes/flask-http-helpers-0167071f46a36e00.yaml (hunk 2 above)
Comment (majorgreys) [reply releasenotes/notes/flask-http-helpers-0167071f46a36e00.yaml:4]: Yep, the link works!

https://449008-61572326-gh.circle-artifacts.com/0/tmp/docs/release_notes.html#deprecation-notes

File: ddtrace/contrib/flask/patch.py (hunk 1 above)
Comment (majorgreys) [reply ddtrace/contrib/flask/patch.py]: Done

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**